controller/backup_target_controller.go

```diff
@@ -0,0 +1,313 @@
+				}
+			}
+		}
+
+		backupTarget.Status.Available = *backupTargetAvailable
+		backupTarget.Status.LastSyncedAt = *backupTargetLastSyncedAt
+		if _, err = btc.ds.UpdateBackupTargetStatus(backupTarget); err != nil && !datastore.ErrorIsConflict(err) {
+			log.WithError(err).Error("Error updating backup target status")
+		}
+	}(&backupTargetAvailable, &backupTargetLastSyncedAt)
+
+	// Check the controller should run synchronization
+	if !backupTarget.Status.LastSyncedAt.IsZero() &&
+		!backupTarget.Status.LastSyncedAt.Before(backupTarget.Spec.SyncRequestAt) {
+		return nil
+	}
+
+	backupTargetClient, err := manager.GenerateBackupTarget(btc.ds)
+	if err != nil {
+		log.WithError(err).Error("Error generate backup target client")
+		backupTargetAvailable = false
+		backupTargetLastSyncedAt = &metav1.Time{Time: time.Now().UTC()}
+		// Ignore error to prevent enqueue
+		return nil
+	}
+
+	// Get a list of all the backup volumes that are stored in the backup target
+	log.Debug("Pulling backup volumes from backup target")
+	res, err := backupTargetClient.ListBackupVolumeNames()
+	if err != nil {
+		log.WithError(err).Error("Error listing backup volumes from backup target")
+		backupTargetAvailable = false
+		backupTargetLastSyncedAt = &metav1.Time{Time: time.Now().UTC()}
+		// Ignore error to prevent enqueue
+		return nil
+	}
+
+	backupStoreBackupVolumes := sets.NewString(res...)
+	log.WithField("BackupVolumeCount", len(backupStoreBackupVolumes)).Debug("Got backup volumes from backup target")
+
+	// Get a list of all the backup volumes that exist as custom resources in the cluster
+	clusterBackupVolumes, err := btc.ds.ListBackupVolume()
+	if err != nil {
+		log.WithError(err).Error("Error listing backup volumes in the cluster, proceeding with pull into cluster")
+	} else {
+		log.WithField("clusterBackupVolumeCount", len(clusterBackupVolumes)).Debug("Got backup volumes in the cluster")
+	}
+
+	clusterBackupVolumesSet := sets.NewString()
+	for _, b := range clusterBackupVolumes {
+		clusterBackupVolumesSet.Insert(b.Name)
+	}
+
+	// Get a list of backup volumes that *are* in the backup target and *aren't* in the cluster
+	// and create the BackupVolume CR in the cluster
+	backupVolumesToPull := backupStoreBackupVolumes.Difference(clusterBackupVolumesSet)
+	if count := backupVolumesToPull.Len(); count > 0 {
+		log.Infof("Found %v backup volumes in the backup target that do not exist in the cluster and need to be pulled", count)
+	}
+	for backupVolumeName := range backupVolumesToPull {
+		log.WithField("backupVolume", backupVolumeName).Info("Attempting to pull backup volume into cluster")
+
+		backupVolumeMetadataURL := backupstore.EncodeMetadataURL("", backupVolumeName, backupTarget.Spec.BackupTargetURL)
+		backupVolumeInfo, err := backupTargetClient.InspectBackupVolumeConfig(backupVolumeMetadataURL)
+		if err != nil || backupVolumeInfo == nil {
+			log.WithError(err).Error("Error getting backup volume metadata from backup target")
+			continue
+		}
```

Comment:
I think this is redundant. BackupVolumeController will be responsible for inspecting it.